lochness/redcap/__init__.py

```diff
@@ -72,117 +80,133 @@ def initialize_metadata(Lochness: 'Lochness object',
+        'fields[0]': redcap_id_colname,
+        'fields[1]': redcap_consent_colname,
+        'events[0]': 'screening_arm_1',
+        'events[1]': 'screening_arm_2',
+        'events[2]': 'baseline_arm_1',
+        'events[3]': 'baseline_arm_2',
+        'filterLogic': f"contains([{redcap_id_colname}],'{site_code_study}')"
+        }
+
 
-    # only pull source_names
-    # mindlamp id is manually added to "chrdig_lamp_id" field
-    field_num = 2
-    for source, (source_name, source_field_name) in \
-            source_source_name_dict.items():
-        record_query[f"fields[{field_num}]"] = source_field_name
+    # to add mindlamp source ID to the record query
+    for num, (source, (source_name, source_field_name)) in \
+            enumerate(source_source_name_dict.items()):
+        record_query[f"fields[{2+num}]"] = source_field_name
 
     # pull all records from the project's REDCap repo
-    try:
-        content = post_to_redcap(api_url,
-                                 record_query,
-                                 f'initializing data {study_name}')
-    except:  # if subject ID field name are not set, above will raise error
-        record_query = {
-            'token': api_key,
-            'content': 'record',
-            'format': 'json',
-        }
-        content = post_to_redcap(api_url,
-                                 record_query,
-                                 f'initializing data {study_name}')
+    content = post_to_redcap(api_url,
+                             record_query,
+                             f'initializing data {study_name}')
 
     # load pulled information as a list of dictionaries
     with tf.NamedTemporaryFile(suffix='tmp.json') as tmpfilename:
         lochness.atomic_write(tmpfilename.name, content)
         with open(tmpfilename.name, 'r') as f:
             data = json.load(f)
 
-    df = pd.DataFrame()
+    # replace empty string as None
+    df = pd.DataFrame(data).replace('', None)
 
-    # extract subject ID and source IDs for each sources
-    for item in data:
-        # filter out data from other sites (if multistudy removed)
-        site_code_redcap_id = item[redcap_id_colname][:2]
-        if site_code_redcap_id != site_code_study:
-            continue
+    # if empty REDCap
+    if len(df) == 0:
+        logger.warn(f'There are no records for {site_code_study}')
+        remove_file_that_may_exist(metadata_study)
+        return
 
-        subject_dict = {'Subject ID': item[redcap_id_colname]}
+    # only keep AMPSCZ rows
+    df = df[df[redcap_id_colname].str.match('[A-Z]{2}\d{5}')]
+
+    # if no data matches AMPSCZ ID
+    if len(df) == 0:
+        logger.warn(f'There are no records for {site_code_study}')
+        remove_file_that_may_exist(metadata_study)
+        return
+
+    # make a single row for each subject record
+    # redcap_event_name column contains timepoint information (different arms)
+    df.drop('redcap_event_name', axis=1, inplace=True)
```

Comment:
`redcap_event_name` must exist in the downloaded JSON:

![image](https://user-images.githubusercontent.com/35086881/175559070-aeca0624-0d0c-49c2-8656-a762899a3ee1.png)

If you dropping will remove it from the JSON, please undo this.